components/draw/InterventionLayer.svelte

```diff
@@ -0,0 +1,67 @@
+<script>
+  import {
+    drawLine,
+    isPolygon,
+    isPoint,
+    isLine,
+    drawCircle,
+    drawPolygon,
+    overwriteSource,
+    overwriteLayer,
+  } from "../../maplibre_helpers.js";
+  import { colors, circleRadius, lineWidth } from "../../colors.js";
+  import { gjScheme, map } from "../../stores.js";
+
+  let source = "interventions";
+
+  overwriteSource($map, source, {
```

Comment:
My head is a little confused by this. 

We define a variable called `source` which we pass into `overwiteSource` here. `overwriteSource` takes 3 args: `map`, `id` and `source`, but we're passing local `source` as the `id` argument.

This is probably just unfamiliarity on my part with the maplibre API and how layers work...

Reply:
The variable `source` is meant as a source ID. It's helpful to name it this way so we can just say `source,` instead of `source: sourceId` when we call `overwriteLayer` later. Should we call the variable `sourceId` maybe to be more clear? Or we used TypeScript, it'd be extremely clear this is being used a string...

MapLibre sources and layers both have a string ID and a definition given by a dictionary.

Reply:
Gotcha. That makes sense I think `sourceId` is clearer but totally get why you've called it `source`.